runtime/arc.js

```diff
@@ -227,8 +227,10 @@ ${this.activeRecipe.toString()}`;
     return id;
   }
 
-  generateID() {
-    return `${this.id}:${this._nextLocalID++}`;
+  generateID(component) {
+    if (component == undefined)
+      component = '';
+    return `${this.id}:${component}${this._nextLocalID++}`;
```

Comment:
why would it be duplicate without the component though?

Reply:
Particle IDs aren't serialized or encoded in recipes. So at the moment the count 'restarts' each time an arc is deserialized. This means that particles can alias handle ids.

This is a temporary fix, once we have a proper Runtime object we can design our ID scheme better.